### PR TITLE
[Frontend][OpenMP] Add functions for checking construct type

### PR DESCRIPTION
**Pull request by kparzysz**
Implement helper functions to identify leaf, composite, and combined constructs.

### CODE REVIEW

**Review by skatrak — APPROVED** [review]
Thank you, based on the current proposal in #87247 this LGTM. If that changes, this will need another review.

File: llvm/unittests/Frontend/OpenMPCompositionTest.cpp [hunk 1]
```diff
@@ -37,3 +37,29 @@ TEST(Composition, GetCompoundConstruct) {
   Directive C6 = getCompoundConstruct({OMPD_parallel_for, OMPD_simd});
   ASSERT_EQ(C6, OMPD_parallel_for_simd);
 }
+
+TEST(Composition, IsLeafConstruct) {
+  ASSERT_TRUE(isLeafConstruct(OMPD_loop));
+  ASSERT_TRUE(isLeafConstruct(OMPD_teams));
+  ASSERT_FALSE(isLeafConstruct(OMPD_for_simd));
```
Comment (skatrak) [inline llvm/unittests/Frontend/OpenMPCompositionTest.cpp:44]: Nit: Can you add one test case for a combined construct? e.g. `ASSERT_FALSE(isLeafConstruct(OMPD_parallel_for));`.

File: llvm/unittests/Frontend/OpenMPCompositionTest.cpp [hunk 2]
```diff
@@ -37,3 +37,29 @@ TEST(Composition, GetCompoundConstruct) {
...
+  ASSERT_FALSE(isLeafConstruct(OMPD_distribute_simd));
+}
+
+TEST(Composition, IsCompositeConstruct) {
+  ASSERT_TRUE(isCompositeConstruct(OMPD_distribute_simd));
+  ASSERT_FALSE(isCompositeConstruct(OMPD_for));
+  ASSERT_TRUE(isCompositeConstruct(OMPD_for_simd));
+  // directive-name-A = "parallel", directive-name-B = "for simd",
+  // only directive-name-A is loop-associated, so this is not a
```
Comment (skatrak) [inline llvm/unittests/Frontend/OpenMPCompositionTest.cpp:53]:
```suggestion
  // only directive-name-B is loop-associated, so this is not a
```

**Review by skatrak — APPROVED** [review]
LGTM, just small comments. Thanks!

File: llvm/unittests/Frontend/OpenMPCompositionTest.cpp (hunk 2 above)
Comment (kparzysz) [reply llvm/unittests/Frontend/OpenMPCompositionTest.cpp]: Done.

**Pull request merged**